llama_cpp/llama.py

```diff
@@ -940,18 +940,47 @@ def _create_completion(
 
         completion_id: str = f"cmpl-{str(uuid.uuid4())}"
         created: int = int(time.time())
+        prefix_token_id: Optional[int] = self.metadata.get("tokenizer.ggml.prefix_token_id")
+        middle_token_id: Optional[int] = self.metadata.get("tokenizer.ggml.middle_token_id")
+        suffix_token_id: Optional[int] = self.metadata.get("tokenizer.ggml.suffix_token_id")
         # If prompt is empty, initialize completion with BOS token to avoid
         # detokenization including a space at the beginning of the completion
         completion_tokens: List[int] = [] if len(prompt) > 0 else [self.token_bos()]
         # Add blank space to start of prompt to match OG llama tokenizer
```

Comment:
What blank space does this comment refer to?

Reply:
It's to avoid a bug when generating from a completely empty prompt that causes a leading space to always be added but I think it may actually be incorrect for non-llama tokenizers, I'll have to check.